src/server.cpp

```diff
@@ -1352,6 +1358,10 @@ void CServer::ProcessData ( const CVector<CVector<int16_t> >& vecvecsData,
             // get a reference to the audio data and gain of the current client
             const CVector<int16_t>& vecsData = vecvecsData[j];
             const double            dGain    = vecdGains[j];
+            const double			dPan = vecdPannings[j];
+            const double			dPanCoefL = sqrt(1-dPan); // faster: (1-dPan)
+            const double			dPanCoefR = sqrt(dPan); // (dPan)
+            // see http://write.flossmanuals.net/csound/b-panning-and-spatialization/ for better formula, if needed
 
             // if channel gain is 1, avoid multiplication for speed optimization
```

Comment:
I check for channel gain == 1 for speed optimiziation but your Pan coefficient is always multiplied. I am not sure how much speed we gain from this "gain == 1" check. Maybe we could do the multiplications also with the dGain always. But I think the less processing we have per block, the lower the jitter will be.

Reply:
I believe that for nowadays computers a multiplication is really very little load. If to think about audio processing and how complex computations are done in audio convolution plugins, for example, I don't think this is really matters much. And if to use panning (if not in mono mode), it needs multiplication anyway.

Reply:
Ok, speed optimization can be done later. That has not to be done in the first place. So it is ok to have additional multiplications for now.